app/services/qa/linked_data/authority_url_service.rb

```diff
@@ -37,29 +42,36 @@ def action_request_variable(action_config, action)
             action_config.qa_replacement_patterns[key]
           end
 
-          def supports_subauthorities?(action_config)
-            action_config.supports_subauthorities?
-          end
-
           def action_subauth_variable(action_config)
             action_config.qa_replacement_patterns[:subauth]
           end
 
-          def action_subauth_variable_value(action_config, subauthority)
-            action_config.subauthorities[subauthority.to_sym]
-          end
-
-          def supports_language_parameter?(action_config)
-            action_config.supports_language_parameter?
+          def action_subauth_variable_value(action_config, request_header)
+            subauth = request_header.fetch(:subauthority, nil)
+            return nil unless subauth && action_config.supports_subauthorities?
+            action_config.subauthorities[subauth.to_sym]
           end
 
           def action_language_variable(action_config)
             action_config.qa_replacement_patterns[:lang]
           end
 
-          def language_value(language)
-            return nil if language.blank?
-            language.first
+          def language_value(action_config, request_header)
+            return nil unless action_config.supports_language_parameter?
+            request_header.fetch(:language, []).first
+          end
+
+          def build_request_header(substitutions, subauthority, language) # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
providing support for deprecated individual parameters

Reply:
is that worth a comment?